controllers/taloscontrolplane_controller.go

```diff
@@ -330,6 +348,7 @@ func (r *TalosControlPlaneReconciler) bootControlPlane(ctx context.Context, clus
 			Bootstrap: capiv1.Bootstrap{
 				ConfigRef: bootstrapRef,
 			},
+			FailureDomain: &failureDomains[rand.Intn(len(failureDomains))],
```

Comment:
this will fail badly if `len(failureDomains) == 0`, should we set it only if it's not empty?